### DISCUSSION TITLE
Update to latest version of Vite?

### DISCUSSION THREAD

**Discussion by nickcotton** [ElMassimo/vite_ruby#113] [Q&A] [answered]
Hi! Wondering the best way to update to latest version of Vite (or even try beta versions)? Do I need to update the gem, `bundle install` and `bundle exec vite install` again? Or would I just update my package.json to the latest vite version? Ideally I'd like to keep any existing config. Thanks!

**Comment by ElMassimo** [comment, marked as answer]
Hi Nick!

`bundle exec vite install` should only be run once when first setting up the gem, and it should not be run when upgrading.

Added a `bin/vite upgrade` command to [3.0.0](https://github.com/ElMassimo/vite_ruby/pull/116), which can be run to perform the following process automatically.

### Upgrading npm packages

You can update the `vite` package independently, but it's always convenient to update `vite-plugin-ruby`, in case there are any breaking changes that require the plugin to be updated as well.

```
yarn upgrade vite vite-plugin-ruby # or npm upgrade
```

You can also specify a version manually in `package.json`, for example: `"vite": "^2.5.0-beta.2",`

It's always convenient to check the changelog when upgrading, specially if upgrading to a major or minor version.

### Upgrading Ruby gems

When upgrading the packages, you might want to also upgrade the gems to stay up-to-date:

```
bundle update vite_ruby vite_rails
```

Upgrading the gems does not upgrade the `npm` packages, and viceversa.

**Reply by nickcotton** [reply]
Fantastic, thanks! This is all really helpful.

**Comment by ElMassimo** [comment]
@nickcotton Added a `bin/vite upgrade` command to [3.0.0](https://github.com/ElMassimo/vite_ruby/pull/116), intended to simplify the process described above.

Would love to get some feedback on it, let me know if you try it!
